### ISSUE TITLE
personSplitter: Keep all cases of the same patient in either a training fold or the test set.

### ISSUE THREAD

**Issue by tomseinen** [OHDSI/PatientLevelPrediction#181]
When training a PLP model and using the randomSplitter, if you use a cohort where one patient can have multiple cases then a number of this patient's cases may end up in a training set fold and the another cases of the same patient can be placed in the test set.

This is, I think, very undesirable, because the test set will have information on the patients in the training set.
The result will be biased, because the model learns which exact patients have a certain outcome.

An example:
> Using a target cohort of all the outpatient visit events of patients older than 60 and an outcome cohort of death events. You want to predict which patients at their outpatient visits will have an event of death within 30 days.
> 
> Using the _randomSplitter_, one visit event with a positive outcome (-10 days before the event) can be placed in the training set while another almost similar visit event from the same patient with also a positive outcome (-29 days before the event) can end up in the test set. The same goes for negative outcomes. 
> When predicting the test set the model will then effortlessly recognise that patient from the training set and classify them correctly.

Would it be possible to do a person split, to keep all the cases of the same patient in one of the folds or test set, before doing a random split, when splitting the dataset?
Or is there a specific reason why that is not implemented? Or not anymore? Because I see that a _personSplitter_ is depreciated. 

Maybe a check to see if all subject_ids are unique can be performed to decide whether to do a person split before the random split?

Thanks!
Tom

**Comment by tomseinen** [comment]
Why apologies, I did not see the subjectSplitter function.

Thank you
Tom
